src/pluto/components/aggregate_report.py

```diff
@@ -0,0 +1,136 @@
+import streamlit as st
+import pandas as pd
+from numerize.numerize import numerize
+import plotly.graph_objects as go
+from src.constants import COLOR_SCHEME
+
+
+class AggregateReport:
+    def __init__(self, data, v1, v2, v3, condo, mapped):
+        self.df = data
+        self.v1 = v1
+        self.v2 = v2
+        self.v3 = v3
+        self.condo = condo
+        self.mapped = mapped
+
+    def __call__(self):
+        st.header("Aggregate Graph")
+
+        self.display_graph()
+
+        st.write(self.aggregate_df.sort_values(by="v", ascending=False))
+
+        st.info(self.info_description)
+
+    @property
+    def aggregate_df(self):
+        return self.df.loc[
+            (self.df.condo == self.condo)
+            & (self.df.mapped == self.mapped)
+            & (self.df.v.isin([self.v1, self.v2, self.v3])),
+            :,
+        ]
+
+    @property
+    def v1_records(self):
+        return self.records_by_version(self.v1)
+
+    @property
+    def v2_records(self):
+        return self.records_by_version(self.v2)
+
+    @property
+    def v3_records(self):
+        return self.records_by_version(self.v3)
+
+    def records_by_version(self, version):
+        return self.aggregate_df.loc[self.aggregate_df.v == version, :].to_dict(
+            "records"
+        )[0]
+
+    @property
+    def y_columns(self):
+        return [
+            "unitsres",
+            "lotarea",
+            "bldgarea",
+            "comarea",
+            "resarea",
+            "officearea",
+            "retailarea",
+            "garagearea",
+            "strgearea",
+            "factryarea",
+            "otherarea",
+            "assessland",
+            "assesstot",
+            "exempttot",
+            "firm07_flag",
+            "pfirm15_flag",
+        ]
+
+    def x_columns(self, v1, v2):
+        return [(v1[i] / v2[i] - 1) * 100 for i in self.y_columns]
+
+    def differences(self, v1, v2):
+        return [v1[i] - v2[i] for i in self.y_columns]
+
+    def version_values(self, version):
+        return [version[i] for i in self.y_columns]
+
+    def version_pair(self, v1, v2):
+        return f"{v1['v']} - {v2['v']}"
```

Comment:
Maybe this is just an aesthetic comment, but these methods are also named after a "thing" and not a "process" (idk if that's a the right term) and don't have the property decorator. Interested to hear your thoughts on why you used the decorator the way you did

Reply:
Generally i think a good rule of thumb is decorator = no function arguments - these are variable and called multiple times. Properties only have one value, so these would change with each call of these functions if that makes sense.

Reply:
Ok yep that does makes sense, sorry I didn't see that when I first reviewed but that is correct I think

Reply:
you could look at the v1_records and v2_records functions for an example of how you might apply a decorator while keeping the underlying logic DRY - they're sort of wrapper shorthand functions for calling a function with a specific argument. Not sure the model applies as well here since there's a lot of versions flying around in these functions, but we could have made a 

```
@property
def v1v2_version_pair(self):
  return version_pair(etc)
```
but it seemed like overkill and not actually improving readability

Reply:
Yea I agree that's overkill. The most readable in my view is still to set the version pair as an attribute when the versions are set, maybe I'll move more logic to init after you're gone and there's no one to stop me lol

Reply:
Haha go for it